enhancements/authentication/direct-use-of-oidc.md

```diff
@@ -0,0 +1,450 @@
+---
+title: direct-use-of-oidc
+authors:
+  - "@stlaz"
+reviewers:
+  - "@s-urbaniak"
+  - "@slaskawi"
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+api-approvers: []
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: []
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Direct use of OIDC as an identity provider
+
+## Summary
+
+When users authenticate to OpenShift, the authentication goes through the `oauth-server`
+which serves as middle-man actor that unifies access to multiple kinds of
+identity providers, such as LDAP, OIDC providers, providers returning HTTP Basic
+Authentication challenges, and similar.
+
+The above differs from vanilla Kubernetes where authentication is dealt directly
+by the `kube-apiserver`. This has limitations especially when it comes to being
+able to login in a uniform manner. On the other hand, `kubectl`, the binary for
+accessing Kubernetes APIs, has evolved through the years to simplify the login
+process to Kubernetes clusters, and communities built useful toolings around
+the binary.
+
+## Motivation
+
+Due to the `oauth-server` being the authentication middle man minting
+OpenShift-specific access tokens, it is impossible to use tokens minted by a
+3rd party OIDC provider directly to authenticate to OpenShift clusters as such
+a token is forwarded directly to the `kube-apiserver` and even if it weren't and
+went directly to the `oauth-server`, there exists no such mechanism to for the
+`oauth-server` to mint an OpenShift-specific access token based on an access or
+ID token from a 3rd party OIDC provider - not to mention that a client should not
+forward its access/ID tokens to a different entity.
+
+For the reasons above, it makes sense to allow configuring OIDC-related flags of
+the `kube-apiserver` directly in order to fully leverage capabilities of the
+community-built tools around OIDC-related authentication. However, there are some
+OpenShift specifics that one needs to bear in mind should they want to use their
+own OIDC provider.
+
+### Goals
+
+- allow directly configuring OIDC-related flags in `kube-apiserver`, guarded
+  by a  `TechPreviewNoUpgrade` feature gate
+- describe expectations that an OIDC provider needs to fulfill in order for its
+  integration with OpenShift to be seamless
+
+### Non-Goals
```

Comment:
is co-existence with the built-in oauth server a goal or a non-goal?  I think suspect a non-goal and in fact we explicitly want to disable the internal oauth-server.  If that's the case, list the disabling as a goal and co-existence as a non-goal